### PR TITLE
Vault_Enterprise_WWW

### PR DESCRIPTION
**Pull request by amanoske**
Adding Vault Enterprise docs to vaultproject.io docs

### CODE REVIEW

**Comment by jefferai** [comment]
This is missing bits to add to the layout. Hit me up at some point and we'll get you an env set up so you can easily see/test the website.

File: website/source/docs/vault-enterprise/index.html.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+sidebar_current: "docs-vault-enterprise"
+description: |-
+  Vault Enterprise features a number of capabilities beyond the open
+  source offering that may be beneficial in certain workflows.
+---
+
+# Vault Enterprise
+
+Vault Enterprise includes a number of features that may be useful in
+specific workflows. These include:
```
Comment (meirish) [inline website/source/docs/vault-enterprise/index.html.md]: @amanoske is there more we should say here?

**Comment by meirish** [comment]
screen shot 2017-02-27 at 6 01 20 pm

File: website/source/docs/http/sys-replication-primary.html.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+---
+
+# /sys/replication/primary/enable
+
+## POST
+
+<dl>
+  <dt>Description</dt>
+  <dd>
+    Enables replication in primary mode. This is used when replication is currently disabled on the cluster
```
Comment (armon) [inline website/source/docs/http/sys-replication-primary.html.md]: I would add a sentence to the effect "Vault replication uses an primary-secondary model, where only one primary should be active at any given time. Multiple primaries may result in data loss."

File: website/source/docs/http/sys-replication-secondary.html.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+  <dt>URL</dt>
+  <dd>`/sys/replication/primary/secondary-token`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">id</span>
+        <span class="param-flags">required</span>
```
Comment (armon) [inline website/source/docs/http/sys-replication-secondary.html.md]: We should document if the `ID` needs to be unique. This is important for `revoke-secondary` since otherwise a non-unique ID might mean we are revoking multiple secondaries.

File: website/source/docs/http/sys-replication-secondary.html.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+
+# /sys/replication/secondary/promote
+
+## POST
+
+<dl>
+  <dt>Description</dt>
+  <dd>
+    Promotes the cluster to become the new primary
```
Comment (armon) [inline website/source/docs/http/sys-replication-secondary.html.md]: I would add the same note about there only being one primary, and that the other primary should be demoted or disabled first.

File: website/source/docs/http/sys-replication.html.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  <dd>`/sys/replication/status`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    None
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+    The printed status of the replication environment.
```
Comment (armon) [inline website/source/docs/http/sys-replication.html.md]: Can we put an example? I'm guess we have a structured return block

File: website/source/docs/vault-enterprise/hsm/behavior.html.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+[here](/docs/internals/security.html)).
+
+When using an HSM, Vault instead stores the master key, encrypted by the HSM,
+into its internal storage. As a result, during an `init` command, the number of
+key shares, threshold, and stored shares are required to be set to `1`, meaning
+to not split the master key, so that the single key share is itself the master
+key. (Vault does not do this automatically as it generally prefers to error
+rather than change parameters set by an operator.)
+
+Vault does not currently support rekeying the master key when protected by an
```
Comment (armon) [inline website/source/docs/vault-enterprise/hsm/behavior.html.md]: Is this true? I think we do support rekey's with HSM, but would be good to confirm with @jefferai

File: website/source/docs/vault-enterprise/hsm/index.html.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+Enterprise](https://www.hashicorp.com/vault.html) that takes advantage of HSMs
+to provide two pieces of special functionality:
+
+ * Master Key Wrapping: Vault protects its master key by transiting it through
+   the HSM for encryption rather than splitting into key shares
+ * Automatic Unsealing: Vault stores its encrypted master key in storage,
+   allowing for automatic unsealing
+
+HSM support is currently limited to devices that support PKCS#11 interfaces and
+provide integration libraries. This includes AWS'
```
Comment (armon) [inline website/source/docs/vault-enterprise/hsm/index.html.md]: I would also include Thales, Utimaco, SafeNet/Gemalto since those have all been tested. We don't want to give people the impression of only CloudHSM

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+redundancy across datacenters.
+
+Prior to Vault 0.7, Vault nodes could only be paired within clusters using a common HA
+storage backend. A single node is locked with a set of other nodes in a cluster to ensure
+high availability. With replication, we intend to allow Vault to allow horizontal 
+scalability between clusters a cross geographically distributed data centers.
+
+## Architecture
+
+When replicating clusters, a single cluster is designed the primary cluster. The
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: s/designed/set as/

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+primary cluster is the active cluster in use within an infrastructure, which
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: "primary cluster is the source of truth, which serves..."

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+high availability. With replication, we intend to allow Vault to allow horizontal 
+scalability between clusters a cross geographically distributed data centers.
+
+## Architecture
+
+When replicating clusters, a single cluster is designed the primary cluster. The
+primary cluster is the active cluster in use within an infrastructure, which 
+serves as the system  of record for replication.
+
+The primary cluster asynchronously replicates data to a series of remote standby
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: This makes the standby's sound like they do nothing. I would rephrase as "The primary cluster replications in near real-time to any number of secondary clusters."

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+The data replicated in Vault 0.7 will include:
+
+ * Secrets
+ * Policies
+ * Configuration details for secret backends
+ * Configuration details for authentication backends
+ * Configuration details for audit backends
+
+Access tokens for secrets are not replicated during the replication process, as tokens are
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: I would also add leases, since leases are managed per cluster.

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+ * Configuration details for audit backends
+
+Access tokens for secrets are not replicated during the replication process, as tokens are 
+local to a cluster that has generated them.  
+
+## Activating Replication 
+
+### Activating the Primary
+
+To activate the primary, run vault write -f sys/replication/primary/enable.
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: I would use code blocks for the commands, so like:

```
$ vault write -f sys/replication/primary/enable
...
```

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+Access tokens for secrets are not replicated during the replication process, as tokens are 
+local to a cluster that has generated them.  
+
+## Activating Replication 
+
+### Activating the Primary
+
+To activate the primary, run vault write -f sys/replication/primary/enable. 
+
+There is currently one optional argument: primary_cluster_addr. This can be used to override
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: You want back ticks for parameters like `primary_cluster_addr`

File: website/source/docs/vault-enterprise/replication/index.html.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+the cluster address that the primary advertises to the secondary, in case the internal network
+address/pathing is different between members of a single cluster and primary/secondary clusters.
+
+### Fetching a Secondary Token
+
+To fetch a secondary bootstrap token, run vault write sys/replication/primary/secondary-token id=<id>. 
+
+The value for ID is opaque to Vault and can be any identifying value you want; this can be used
```
Comment (armon) [inline website/source/docs/vault-enterprise/replication/index.html.md]: I would recommend unique ID's (even if not required) so that you can uniquely identify which secondary to revoke access for.

File: website/source/docs/http/sys-replication-secondary.html.md (hunk 3 above)
Comment (jefferai) [inline website/source/docs/http/sys-replication-secondary.html.md]: Correct, it needs to be unique (and this will be enforced by the primary).

File: website/source/docs/vault-enterprise/hsm/behavior.html.md (hunk 6 above)
Comment (jefferai) [inline website/source/docs/vault-enterprise/hsm/behavior.html.md]: "It's tricky". When the primary is protected by an HSM we do not support rekeys; however, when a secondary is HSM protected we do, because basically the secondary will do the right thing when a rekey happens.
